x/mongo/driver/connstring/connstring.go

```diff
@@ -245,7 +246,7 @@ func (p *parser) parse(original string) error {
 		if strings.Contains(password, "/") {
 			return fmt.Errorf("unescaped slash in password")
 		}
-		p.Password, err = url.QueryUnescape(password)
+		p.Password, err = url2.Unescape(password, url2.EncodeUserPassword)
```

Comment:
Same as above, update this to call `url.PathUnescape` instead of `url2.Unescape`.